js/src/admin/components/DismissedFlagsTable.js

```diff
@@ -0,0 +1,84 @@
+import Button from 'flarum/components/Button';
+import Component from 'flarum/Component';
+import LoadingIndicator from 'flarum/components/LoadingIndicator';
+import DismissedFlagsModal from './DismissedFlagsModal';
+import humanTime from 'flarum/helpers/humanTime';
+import username from 'flarum/helpers/username';
+import avatar from 'flarum/helpers/avatar';
+
+export default class DismissedFlagsTable extends Component {
+  init() {
+    this.data = [];
+    this.loading = true;
+
+    app.store.find('flags', { dismissed: true, sort: 'dismissedAt' }).then(res => {
+      this.data = res.payload.data;
+      this.loading = false;
+
+      m.redraw();
+    });
+  }
+
+  view() {
+    if (this.loading) {
+      return <LoadingIndicator />;
+    }
+
+    return (
+      <div className="DismissedFlagsTable">
+      <table className="DismissedFlagsTable-results">
+          <thead>
+            <th>{app.translator.trans('flarum-flags.admin.dismissed.fields.username')}</th>
+            <th>{app.translator.trans('flarum-flags.admin.dismissed.fields.discussion')}</th>
+            <th>{app.translator.trans('flarum-flags.admin.dismissed.fields.created_at')}</th>
+            <th>{app.translator.trans('flarum-flags.admin.dismissed.fields.dismissed_at')}</th>
+            <th>{app.translator.trans('flarum-flags.admin.dismissed.fields.dismissed_by')}</th>
```

Comment:
We could perhaps turn this into
```js
['username', 'discussion', 'created_at', ...]
  .map(k => <th>...</th>)
```

Maybe even extensible - add a method that returns the keys in an ItemList or regular array and loop them for columns ?

Could perhaps do the same with the tbody below.

Not sure if we want this at all, just throwing ideas.

Reply:
I would go against the mapping idea. In my opinion, it doesn't make sense to have the columns in the header row dynamically generated from an array when the columns in the table body rows are hard defined. I think we'd end up creating more overhead/confusion for ourselves.

One thing to note is the final column in the header row is empty, therefore unless we decide that the final column should have a label we'd end up wiring some logic to skip over rendering any text output. I feel this suggestion is finding a solution to an issue which is not there. Also, this page on the admin panel was supposed to be a [short term solution](https://github.com/flarum/flags/pull/21#issuecomment-596014300).

Reply:
Well that's why I said perhaps make the tbody extensible as well. If it's a temporary solution tho, it's probs fine then.

Also - you could have an empty <th> after the mapped array (that wouldn't exclude it).